File: mvc-sample/src/main/kotlin/suzp1984/github/io/jbig_kotlin/states/BaseState.java
package suzp1984.github.io.jbig_kotlin.states;

import suzp1984.github.io.jbig_kotlin.controller.MainController;

/**
 * Created by suzhenxi on 11/28/2016.
 */

public interface BaseState {

    MainController.TabItem getSelectedTabItem();

    void setSelectedTabItem(MainController.TabItem item);

    void registerForEvents(Object object);

    void unregisterForEvents(Object object);
}

File: mvc-sample/src/main/kotlin/suzp1984/github/io/jbig_kotlin/states/BaseState.kt
package suzp1984.github.io.jbig_kotlin.states

import suzp1984.github.io.jbig_kotlin.controller.MainController

/**
 * Created by suzhenxi on 11/28/2016.
 */

interface BaseState {

    var selectedTabItem: MainController.TabItem?

    fun registerForEvents(`object`: Any)

    fun unregisterForEvents(`object`: Any)
}
